bids_to_msd_dataset_conversion.py

```diff
@@ -0,0 +1,290 @@
+params["labels"] = {
+    "0": "background",
+    "1": "sc-lesion"        # TODO: Define the classes to be be segmented
+    }
+params["license"] = "xxx"
+params["modality"] = {
+    "0": "MRI"
+    }
+params["name"] = "sci-zurich"       # TODO: Add the name of the dataset
+params["reference"] = "N/A"
+params["tensorImageSize"] = "3D"
+
+subjects_dict = {
+    "training": train_subjects,
+    "validation": val_subjects,
+    "test": test_subjects
+}
+
+# loop through the training, validation and test splits and create dictionaries that contain the 
+# (image, label) pairs for each subject
+for name, subs_list in subjects_dict.items():
+
+    temp_list = []
+    for subject_no, subject in enumerate(tqdm(subs_list, desc='Loading Volumes')):
+
+        # recursively get all the files for the subject
+        files = sorted(glob.glob(os.path.join(args.path_data, subject) + "/**/*.nii.gz", recursive=True))
+
+        temp_data = {}
+        session_ctr, contrast_ctr = 0, 0
+        for file in files:
+            # build file names
+            subjectID, sessionID, contrast_suffixID, datatype, filename = utils.fetch_subject_info(file)
+            # print(subjectID, sessionID, contrast_suffixID, datatype, filename)
+
+            if sessionID == '':
+                # print("No session ID found, possibly cross-sectional dataset. Moving on to check for grouping by contrasts....")
+
+                if args.group_by_contrasts:
+                    # print("Found that --group_by_contrasts is set. Moving on to check for common label suffix....")
+
+                    # check if args.common_label is specified. If not, raise an error
+                    assert args.common_label_contrast != '', "Please specify the common label contrast using the --common_label_contrast argument. " \
+                                                                "This will be used to pair all the contrasts with the same label file."
+
+                    # check if only the specified contrasts have to be included
+                    if args.include_contrasts is not None and contrast_suffixID in args.include_contrasts:
+                        image_file = os.path.join(root, subjectID, datatype, filename)
+
+                        if contrast_suffixID == args.common_label_contrast:
+                            label_file = os.path.join(root, 'derivatives', 'labels', subjectID, datatype, utils.add_suffix(filename, args.label_suffix))
+
+                        # # TODO: check if there are missing contrasts
+
+                    elif args.include_contrasts is None:
+                        # assuming all contrasts have to be included
+                        image_file = os.path.join(root, subjectID, datatype, filename)
+                        if contrast_suffixID == args.common_label_contrast:
+                            label_file = os.path.join(root, 'derivatives', 'labels', subjectID, datatype, utils.add_suffix(filename, args.label_suffix))
```

Comment:
`'derivatives', 'labels'` is hardcoded several times across the script. It would be better to define it as a global variable at the beginning of the script, e.g.:

```
PATH_DERIVATIVES = os.path.join('derivatives', 'labels')
```

This will allow easy modification in case the user uses, for example, `derivatives/manual_labels` instead of `derivatives/labels`